### ISSUE TITLE
Problem with the wizard's script

### ISSUE THREAD

**Issue by Felipefpl** [Gibberlings3/EE_Fixpack#76]
If you select a script to make the wizards attack he/she always attack using the weakest spells while leaving the strong spells intact, i found that really annoying and it doesnt make any sense at all, the correct would be to make them attack using always the strongest INDIVIDUAL spells first followed by the 2nd strongest individual spells and so on. They NEVER must use the AOE's spells since it causes casualties to our side.

**Comment by szaumoor** [comment]
Just out of curiosity, how do you determine what spell is the strongest in a battle?

**Comment by Felipefpl** [comment]
By its level of course. I'm curious: why do you ask?

**Comment by szaumoor** [comment]
Because the level of the spell is quite often not very indicative of how effective it will be in a particular battle. There are various factors to take into account. So I'm wondering how that sort of thing would be computed (and how intelligent the script can be).

**Comment by Felipefpl** [comment]
Sorry but i disagree, it's AWFUL when you are fighting enemies that are level 10 and above and your mage keeps hitting them with magic missiles instead of something stronger that does more damage.

**Comment by szaumoor** [comment]
It's not as simple as saying "higher level equals more effective". Magic Missile is in fact an extremely good spell, even compared to level 9 spells such as Energy Drain, in almost all circumstances.

I think it would be good to make the script more intelligently effective, but I think it's overly simplistic to put it in these terms.

Regardless, I don't think this is a bug (just my opinion) unless you can show it's doing something that is clearly unintended originally. After all, this mod is an attempt at fixing bugs, not improving features.

**Comment by Roberciiik** [comment]
You should know that you can "take control" over your mage and cast a spell that you like. ;)

I never use Wizard/Priest AI scripts (and never would) to spell 9th on a bunch of goblins.

As szaumoor said, in this case the script should take care of immunities and so on, because maybe magic missiles are better than a Comet if someone is protected from fire.

**Comment by Felipefpl** [comment]
> It's not as simple as saying "higher level equals more effective". Magic Missile is in fact an extremely good spell, even compared to level 9 spells such as Energy Drain, in almost all circumstances.

Well, this depends of what spells you choose. ;)
 
> I think it would be good to make the script more intelligently effective, but I think it's overly simplistic to put it in these terms.

Indeed, a better script would be more effective.

> Regardless, I don't think this is a bug (just my opinion) unless you can show it's doing something that is clearly unintended originally. After all, this mod is an attempt at fixing bugs, not improving features.

I disagree, several things here arent bugfixes only, there are a lot of improvements https://www.gibberlings3.net/subpages/ee-fixpack/docs_bgee/

**Comment by Felipefpl** [comment]
> You should know that you can "take control" over your mage and cast a spell that you like. ;)

Yes but that' not the point of this post, the point is show an oversight of the devs.

> I never use Wizard/Priest AI scripts (and never would), especially to cast 9th level spell on a bunch of goblins.

That's you but of course i wasnt facing goblins...
 
> As szaumoor said, in this case the script should take care of immunities and so on, because maybe magic missiles are better than a Comet if someone is protected from fire. Which is harder to implement if it has to be smart AI script. :)

To each his own, if you both like things this way that's ok.

**Comment by Roberciiik** [comment]
I was kidding a bit. Nevertheless, I feel like those scripts are simple help if someone can't control all characters at once. I'm not sure if the scripts tend to be on the Player intelligence. 

If there is a room for improvement, it could be fixed, surely.
